app/scripts/modules/docker/src/pipeline/trigger/DockerTriggerTemplate.tsx

```diff
@@ -128,49 +153,77 @@ export class DockerTriggerTemplate extends React.Component<
   };
 
   public render() {
-    const { tags, tagsLoading, loadError, selectedTag } = this.state;
+    const { digest, tags, tagsLoading, loadError, selectedTag, lookupType } = this.state;
 
     const options = tags.map(tag => {
       return { value: tag } as Option<string>;
     });
 
     return (
-      <div className="form-group">
-        <label className="col-md-4 sm-label-right">Tag</label>
-        {tagsLoading && (
-          <div className="col-md-6">
-            <div className="form-control-static text-center">
-              <Spinner size={'small'} />
-            </div>
+      <>
+        <div className="form-group">
+          <div className="sm-label-right col-md-4">Type</div>
+          <div className="col-md-3">
+            <TetheredSelect
+              clearable={false}
+              value={lookupType}
+              options={[{ value: 'digest', label: 'Digest' }, { value: 'tag', label: 'Tag' }]}
```

Comment:
no idea whether this component avoids renders when its props are the same, but if so could pull this array up to a file-level constant.